### PR TITLE
[SPARK-31286][SQL][DOC] Specify formats of time zone ID for JSON/CSV option and from/to_utc_timestamp

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to update the doc for the `timeZone` option in JSON/CSV datasources and for the `tz` parameter of the `from_utc_timestamp()`/`to_utc_timestamp()` functions, and to restrict format of config's values to 2 forms:
1. Geographical regions, such as `America/Los_Angeles`.
2. Fixed offsets - a fully resolved offset from UTC. For example, `-08:00`.

### Why are the changes needed?
Other formats such as three-letter time zone IDs are ambitious, and depend on the locale. For example, `CST` could be U.S. `Central Standard Time` and `China Standard Time`. Such formats have been already deprecated in JDK, see [Three-letter time zone IDs](https://docs.oracle.com/javase/8/docs/api/java/util/TimeZone.html).

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
By running `./dev/scalastyle`, and manual testing.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@HyukjinKwon @cloud-fan Please, have a look at the PR.

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala [hunk 1]
```diff
@@ -198,8 +209,19 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
    *
    * You can set the following option(s):
    * <ul>
-   * <li>`timeZone` (default session local timezone): sets the string that indicates a timezone
-   * to be used to format timestamps in the JSON/CSV datasources or partition values.</li>
+   * <li>`timeZone` (default session local timezone): sets the string that indicates a time zone ID
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: is the `timezone` option effective for writing? AFAIK it's for parsing only.

Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: If a pattern contains sub-pattern for time zone like `V` or `z`, which one formatter should output?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: ah got it

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: shall we update the comment? It's not for parsing.

Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: Updated

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Pull request closed without merging**